module-2/backend/week_four.md

```diff
@@ -10,14 +10,49 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+
+A way to store data on a user's browser
+
 * What’s the difference between a session and a cookie?
+
+Session data is stored on the server side, while cookies are stored on the user's side.
+
 * What’s a flash and when do you want to use flashes?
+
+A message that is displayed on the page after an action. Logging in, logging out, wrong credentials, account created, item added to cart, removed from cart, etc.
+
 * Why do people say “HTTP is stateless”?
+
+Because HTTP does not hold state. To get around this we need to save information so that we can send it in every request/response that it is needed in.
+
 * What’s authentication? Explain.
+
+Verifying inputted information such as a username and password combination against the combination that is stored on the server.
+
 * What’s the difference between authentication and authorization?
+
+Authentication is done to verify things such as a user, authorization is done to ensure that access to correct parts of our applications are shown and are accessible to users.
+
 * What’s a before filter?
+
+A before filter allows us to declare which methods on our controller can only be called depending on the condtions that we set, such as calling edit and update.
```

Comment:
Can also use a `before_action` to run any method before other methods in our controller. E.g. our new `set_backpack` method doesn't necessarily limit access to methods on our controller, but still runs before each one.